src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5012,25 +5012,50 @@ static QVariant fcnGetFeature( const QVariantList &values, const QgsExpressionCo
   {
     return QVariant();
   }
-
-  QString attribute = QgsExpressionUtils::getStringValue( values.at( 1 ), parent );
-  int attributeId = featureSource->fields().lookupField( attribute );
-  if ( attributeId == -1 )
+  QgsFeatureRequest req;
+  QString cacheValueKey;
+  if ( values.at( 1 ).type() == QVariant::Map )
   {
-    return QVariant();
+    QVariantMap attributeMap = QgsExpressionUtils::getMapValue( values.at( 1 ), parent );
+
+    QMap <QString, QVariant>::const_iterator i = attributeMap.constBegin();
+    QString filterString;
+    for ( ; i != attributeMap.constEnd(); ++i )
+    {
+      if ( !filterString.isEmpty() )
+      {
+        filterString.append( " AND " );
+      }
+      filterString.append( QStringLiteral( "( %1 = %2 )" ).arg( QgsExpression::quotedColumnRef( i.key() ),
+                           QgsExpression::quotedString( i.value().toString() ) ) );
```

Comment:
Does this only work with strings? Or should that be quotedValue?

Reply:
I simply used the pattern that was already present in the function, I didn't look into changing such things to not break anything. I can change it on this line and below if needed.

Reply:
Ideally, that would be using `QgsExpression::createFieldEqualityExpression`

Reply:
If you could add a test for that with an integer or even a date value, that would be nice

Reply:
@m-kuhn Sure thing, I'll look into it ASAP!